File: cell_discharge_data.py
import numpy as np
import pandas as pd


class CellDischargeData:
    """
    Battery cell data from discharge test.
    """

    def __init__(self, path):
       
        data_frame = pd.read_csv(path, sep="\t")
        self.time = data_frame.iloc[:,0]
        self.voltage = data_frame.iloc[:,1]
        self.current = data_frame.iloc[:,2]

  
    @classmethod
    def process(cls, path):
        """
        Process the original discharge data for one section. This section of
        data is used for model development.
        """
        data = cls(path)

        data_frame = pd.read_csv(path, sep="\t")
          

        data.time  = data_frame.iloc[:,0]
        data.voltage = data_frame.iloc[:,1]
        data.current = data_frame.iloc[:,2]
      
        return data

    @classmethod
    def process_discharge_only(cls, path):
        """
        Process the original discharge data for just the discharge portion.
        """
        data = cls(path)
        data_frame = pd.read_csv(path, sep="\t")
          
        data.time  = data_frame.iloc[:,0]
        data.voltage = data_frame.iloc[:,1]
        data.current = data_frame.iloc[:,2]
                     
        return data

File: CellDischargeData.m
classdef CellDischargeData

    % battery discharge data %
    
    properties
        time
        voltage
        current
    end
    
    methods
        function obj = CellDischargeData(path)
            % 생성자: 파일에서 데이터를 읽어와 객체의 속성을 초기화
            data = readtable(path, 'Delimiter', '\t');
            obj.time = data{:, 1};
            obj.voltage = data{:, 2};
            obj.current = data{:, 3};
        end
    end
    
    methods (Static)
        function data = process(path)
            % 원본 방전 데이터의 한 구간을 처리하는 정적 메서드
            data = CellDischargeData(path);
        end
        
        function data = process_discharge_only(path)
            % 방전 데이터의 방전 부분만 처리하는 정적 메서드
            data = CellDischargeData(path);
        end
    end
end

